src/pch/system.h

```diff
@@ -12,9 +12,8 @@
 #include <string>
 #include <algorithm>
 #include <memory>
+#include <functional>
```

Comment:
I added `<set>` since it's now in `ct_defs.h` for the sake of being able to add the Cantera-wide `using`. However, I didn't see any compilation speedup for adding the other two headers here.

I expect to give `std::any` a shot in a future PR also making use of some other C++17 features. Perhaps it would be worth checking again at that point whether including `<any>` in the PCH helps at all.